### PR TITLE
Move subadmin middleware to app

### PR DESCRIPTION
**Pull request by DeepDiver1975**

### CODE REVIEW

File: lib/AppInfo/Application.php [hunk 1]
```diff
@@ -0,0 +1,32 @@
+<?php
+/**
+ * Created by PhpStorm.
```
Comment (phil-davis) [inline lib/AppInfo/Application.php]: should have a standard ownCloud header?

**Comment by phil-davis** [comment]
Working from a UI POV:
- login as admin, create auser in admin group, buser unpriv
- login as auser, try to go to Users page, works. add a user, put in a group... - works.
- login as buser, try to go to Users page, "access forbidden"

So unpriv users cannot get to the webUI users page.

Is the Users page using some different endpoint underneath? That would need to be check to make sure that an an unpriv user cannot POST transactions to that endpoint "manually".

Somehow we should run ``core`` API test features like auth and provisioning-v1 with the user_management app installed and enabled. That will confirm that user/group creation... still "behaves itself" when the user_management app is enabled.

**Comment by DeepDiver1975** [comment]
> Is the Users page using some different endpoint underneath? That would need to be check to make sure that an an unpriv user cannot POST transactions to that endpoint "manually".

the app has own routes to interact with - see the routes.php file

**Comment by phil-davis** [comment]
As an admin, go to:
```
http://localhost:8080/index.php/apps/user_management/users
```
JSON of user data is returned.

Do the same as unpriv user => access forbidden

Good.

**Pull request merged**